packages/puppeteer-core/src/bidi/core/Session.ts

```diff
@@ -105,6 +105,19 @@ export class Session
     browserEmitter.once('closed', ({reason}) => {
       this.dispose(reason);
     });
+
+    // TODO: Currently, some implementations do not emit navigationStarted event
+    // for fragement navigations (as per spec) and some do. This could emits a
+    // synthetic navigationStarted to work around this inconsistency.
+    const seen = new WeakSet();
+    this.on('browsingContext.fragmentNavigated', info => {
```

Comment:
We could but 1) it does not affect Chrome because it already handles duplicate events 2) we will fix Chrome to behave similarly as Firefox so it would allow us to roll changes more easily.